Expensify/App
# Add auto-update on hide workflow for Electron on Mac

### Details
This change allows the Electron app to auto-update when it has finished downloading the update and it is backgrounded.

### Fixed Issues
<Please replace GH_LINK with the link to the GitHub issue this Pull Request is fixing>
Fixes #1011.

### Tests

1. Install and run MinIO locally.
2. Point the `publish` spec in `electron.config.js` at the local server.
3. Publish a new version with the build number set very high (i.e. `1.0.1-999`).
4. Build a new version for local development with a lower build number (i.e. `1.0.1-374`). It is important that this is notarized, so you must use `npm run desktop-build`. This build should also have the `publish` spec pointed at MinIO.
5. Run the app.
6. Verify the app version is `1.0.1-374`.
7. Once the new update has been downloaded, either hide or close the window.
8. The app may reboot a few times.
9. Foreground the app and verify that the app version is `1.0.1-999`.

### Tested On


### Screenshots


#### Web

N/A

#### Mobile Web

N/A

#### Desktop

Video of the experience: https://www.dropbox.com/s/17jhevgcax2xz6x/Screen%20Recording%202021-02-24%20at%2010.13.15%20PM.mov?dl=0

Update modal:
<img width="1312" alt="image" src="https://user-images.githubusercontent.com/993682/109097696-d8b50200-76ed-11eb-9032-9f79e9940ed6.png">



#### iOS

N/A

#### Android

N/A
